redis_json/src/key_value.rs

```diff
@@ -422,45 +406,32 @@ impl<'a, V: SelectValue + 'a> KeyValue<'a, V> {
     }
 
     pub fn is_equal<T1: SelectValue, T2: SelectValue>(a: &T1, b: &T2) -> bool {
-        match (a.get_type(), b.get_type()) {
-            (SelectValueType::Null, SelectValueType::Null) => true,
-            (SelectValueType::Bool, SelectValueType::Bool) => a.get_bool() == b.get_bool(),
-            (SelectValueType::Long, SelectValueType::Long) => a.get_long() == b.get_long(),
-            (SelectValueType::Double, SelectValueType::Double) => a.get_double() == b.get_double(),
-            (SelectValueType::String, SelectValueType::String) => a.get_str() == b.get_str(),
-            (SelectValueType::Array, SelectValueType::Array) => {
-                if a.len().unwrap() == b.len().unwrap() {
-                    for (i, e) in a.values().unwrap().enumerate() {
-                        if !Self::is_equal(e, b.get_index(i).unwrap()) {
-                            return false;
-                        }
-                    }
-                    true
-                } else {
-                    false
+        a.get_type() == b.get_type()
+            && match a.get_type() {
+                SelectValueType::Null => true,
+                SelectValueType::Bool => a.get_bool() == b.get_bool(),
+                SelectValueType::Long => a.get_long() == b.get_long(),
+                SelectValueType::Double => a.get_double() == b.get_double(),
+                SelectValueType::String => a.get_str() == b.get_str(),
+                SelectValueType::Array => {
+                    a.len().unwrap() == b.len().unwrap()
+                        && a.values()
+                            .unwrap()
+                            .zip(b.values().unwrap())
+                            .try_for_each(|(a, b)| Self::is_equal(a, b).then_some(()))
```

Comment:
I believe the same result can be achieved using the `any` method of the `Iterator` and be a little simpler:

```rust
                    a.len().unwrap() == b.len().unwrap()
                        && !a.values()
                            .unwrap()
                            .zip(b.values().unwrap())
                            .try_for_each(|(a, b)| Self::is_equal(a, b).then_some(()))
                            .any(|(a, b)| !Self::is_equal(a, b))
```

Reply:
`all`